systems/robotInterfaces/+footstepPlanner/nonlinearCollocation.m

```diff
@@ -295,14 +295,15 @@
   c = zeros(nvar, 1);
 
   w_goal = diag(weights.goal);
-  if seed_plan.footsteps(end).frame_id == biped.foot_frame_id.right
-    xg = reshape(goal_pos.right, [], 1);
-  else
-    xg = reshape(goal_pos.left, [], 1);
+  for j = (nsteps - 1):nsteps
```

Comment:
The plans always include the current poses of the feet as the first two steps, so a plan always has at least three steps, even when max_num_steps is 1. In that case, this change should have no effect on the result, but it also shouldn't cause a problem